pyodata/v2/model.py

```diff
@@ -840,135 +955,165 @@ def from_etree(schema_nodes):
             decl = Schema.Declaration(namespace)
             schema._decls[namespace] = decl
 
-            for enum_type in schema_node.xpath('edm:EnumType', namespaces=NAMESPACES):
-                etype = EnumType.from_etree(enum_type, namespace)
+            for enum_type in schema_node.xpath('edm:EnumType', namespaces=config.namespaces):
+                try:
+                    etype = EnumType.from_etree(enum_type, namespace, config)
+                except (PyODataParserError, AttributeError) as ex:
+                    config.err_policy(ParserError.ENUM_TYPE).resolve(ex)
+                    etype = NullType(enum_type.get('Name'))
+
                 decl.add_enum_type(etype)
 
-            for complex_type in schema_node.xpath('edm:ComplexType', namespaces=NAMESPACES):
-                ctype = ComplexType.from_etree(complex_type)
+            for complex_type in schema_node.xpath('edm:ComplexType', namespaces=config.namespaces):
+                try:
+                    ctype = ComplexType.from_etree(complex_type, config)
+                except (KeyError, AttributeError) as ex:
+                    config.err_policy(ParserError.COMPLEX_TYPE).resolve(ex)
+                    ctype = NullType(complex_type.get('Name'))
+
                 decl.add_complex_type(ctype)
 
-            for entity_type in schema_node.xpath('edm:EntityType', namespaces=NAMESPACES):
-                etype = EntityType.from_etree(entity_type)
+            for entity_type in schema_node.xpath('edm:EntityType', namespaces=config.namespaces):
+                try:
+                    etype = EntityType.from_etree(entity_type, config)
+                except (KeyError, AttributeError) as ex:
+                    config.err_policy(ParserError.ENTITY_TYPE).resolve(ex)
+                    etype = NullType(entity_type.get('Name'))
+
                 decl.add_entity_type(etype)
 
         # resolve types of properties
         for stype in itertools.chain(schema.entity_types, schema.complex_types):
-            if stype.kind == Typ.Kinds.Complex:
+            if not isinstance(stype, NullType) and stype.kind == Typ.Kinds.Complex:
```

Comment:
Please, do not join unrelated expressions via logic operators.

Reply:
danone